### PR TITLE
266 Generalize Data Manifest Tool For HuBMAP and SenNet

### PR DESCRIPTION
**Pull request by tjmadonna**
`clt/app.cfg` required to build a distribution. Example cfg files for HuBMAP and SenNet can be found in `clt/`.

### CODE REVIEW

**Review by DerekFurstPitt — APPROVED** [review]
Everything looks right to me.

**Review by yuanzhou — COMMENTED** [review]
@tjmadonna I've never used this CLT tool so I didn't really run it to check things out. Just some comments after looking at this PR:

- We probably don't need the `dev-integrate` branch in this repo. Not like other APIs we do the initial integration tests on DEV. So this PR can go to `main` instead if we test against your branch directly. What's your thought @shirey?
- Since we publish it to PyPI and considering this work will be shared, `pyproject.toml` is the new standardized project settings file should be used to replace the old `setup,py` and you can specify the build tools, along with other benefits. 
- I still see "hubmap" being mentioned at https://github.com/x-atlas-consortia/clt/blob/266-generalize-data-manifest-tool/clt/__main__.py#L55 and https://github.com/x-atlas-consortia/clt/blob/266-generalize-data-manifest-tool/clt/__main__.py#L200, even just comments, nice to keep things consistent with the generalization work.

**Comment by yuanzhou** [comment]
From Tyler:
>I saw your comment on the CLT PR and totally agree that we should be using the pyproject.toml. Currently the requirement is that we have two packages on PyPI from the same code (with different configurations). I am not sure how to do this with pyproject.toml as most of the values in that file have to be static. My initial thought it to generate a pyproject.toml file during the build process based on the configuration file. Any thoughts or any other ideas on this?

From Zhou:
>Hi Tyler, I took a closer look, essentially we don't need the config file.
```
# Below info can be part of the README documentation, no need to show via the CLT commands
INSTALLATION_DOCS_URL = "" # I don't see it being used in the code
MANIFEST_DOCS_URL = "" # I don't see it being used in the code
USAGE_DOCS_URL = ""
EXAMPLE_ENTITY_ID = ""

# Metadata of the package can go into pyproject.toml directly
PKG_NAME = "x-atlas-consortia-clt"
PKG_VERSION = "1.1.3"
PKG_AUTHOR = "Atlas Consortia"
PKG_AUTHOR_EMAIL = "api-developers@hubmapconsortium.org" # We can update later once set up a new email for x-atlas-consortia

# Below items can be input options/parameters in the CLT commands
ENTITY_ID_NAME = ""
INGEST_WEBSERVICE_URL = ""
```

File: src/atlas_consortia_clt/common/help.py
Comment (yuanzhou) [inline src/atlas_consortia_clt/common/help.py:18]: I didn't test the `transfer` subcommand because I don't have the proper Globus setup. It's unclear to me if I should use `manifest-file` (described here) or `manifest` (specified in the code: https://github.com/x-atlas-consortia/clt/blob/266-generalize-data-manifest-tool/src/atlas_consortia_clt/common/main.py#L22), please make them consistent.

File: src/atlas_consortia_clt/common/help.py
Comment (yuanzhou) [inline src/atlas_consortia_clt/common/help.py:57]: Typo `mages of files`

File: src/atlas_consortia_clt/common/main.py
Comment (yuanzhou) [inline src/atlas_consortia_clt/common/main.py:22]: See above, `manifest-file` V.S. `manifest`

File: README.md
Comment (yuanzhou) [inline README.md:15]: The correct spelling is `ORCID`

**Review by yuanzhou — CHANGES_REQUESTED** [review]
@tjmadonna Great work with restructuring the project and thanks for improving the documentation! 
- I changed the base branch of this PR to `main` and deleted `dev-integrate` since we don't use a DEV environment to test this CLT. Also FYI @DerekFurstPitt since you created that branch.
- When I tested the `login | logout` commands, if I use `login` for a second time when I have already logged in, it still opens the browser and asks me to login again. Same for `logout` twice. The CLT doesn't seem to maintain the status. Totally fine, just wanted to point out.
- A few minor tweaks requested along the lines of code. Once addressed, we can merge this PR if @maxsibilla also approves.

**Review by yuanzhou — APPROVED** [review]

**Comment by maxsibilla** [comment]
@tjmadonna overall this looks great. 2 things I would recommend doing:
1) Under the `Development` section of the README just clarify a little more that the user needs to copy and overwrite `app.cfg.example` into `app.cfg` and then then need to build the .toml with `python3 -m pip install .` prior to running any commands. Its good to link to the Python documentation but adding these additional steps just allows developers to use this faster.
2) Make sure we get your branch in the documentation repo into `main` so that this resolves: https://docs.sennetconsortium.org/clt

**Review by maxsibilla — APPROVED** [review]

**Pull request merged**
